cmd/server/assets/codes/bulk-issue.html

```diff
@@ -282,33 +308,37 @@
         request["tzOffset"] = tzOffset;
         return request
       }
-    });
 
-    function uploadBatch(data) {
-      return $.ajax({
-        url: '/codes/issue',
-        type: 'POST',
-        dataType: 'json',
-        cache: false,
-        contentType: 'application/json',
-        headers: { 'X-CSRF-Token': '{{.csrfToken}}' },
-        // TODO: at the moment we only support one-at-a-time
-        data: JSON.stringify(data[0]),
-        success: function(result) {
-          total++;
-          if (result.error) {
-            flash.error(result.error);
-          }
-        },
-        error: function(xhr, resp, text) {
-          let message = resp;
-          if (xhr && xhr.responseJSON && xhr.responseJSON.error) {
-            message = message + ": " + xhr.responseJSON.error;
-          }
-          flash.error(message);
-        },
-      });
-    }
+      function uploadBatch(data, i) {
+        return $.ajax({
+          url: '/codes/issue',
+          type: 'POST',
+          dataType: 'json',
+          cache: false,
+          contentType: 'application/json',
+          headers: { 'X-CSRF-Token': '{{.csrfToken}}' },
+          // TODO: at the moment we only support one-at-a-time
+          data: JSON.stringify(data[0]),
+          success: function(result) {
+            total++;
+            if (result.error) {
+              flash.error(result.error);
+            }
+          },
+          error: function(xhr, resp, text) {
+            let message = resp;
+            if (xhr && xhr.responseJSON && xhr.responseJSON.error) {
+              message = message + ": " + xhr.responseJSON.error;
+            }
+            $errorDiv.removeClass('d-none');
+            let row = "<tr><td>" + i + "</td><td>" + data[0]["phone"] + "</td><td>" +
```

Comment:
This is vulnerable to reflected XSS. Build the row by hand and use jQuery's `.text()`.